shotover/src/transforms/kafka/sink_cluster/shotover_node.rs

```diff
@@ -33,9 +40,85 @@ pub(crate) struct ShotoverNode {
     state: Arc<AtomicShotoverNodeState>,
 }
 
+impl ShotoverNode {
+    #![allow(unused)]
+    pub(crate) fn is_up(&self) -> bool {
+        self.state.load(Ordering::Relaxed) == ShotoverNodeState::Up
+    }
+
+    pub(crate) fn set_state(&self, state: ShotoverNodeState) {
+        self.state.store(state, Ordering::Relaxed)
+    }
+}
+
 #[atomic_enum]
 #[derive(PartialEq)]
 pub(crate) enum ShotoverNodeState {
     Up,
     Down,
 }
+
+pub(crate) fn start_shotover_peers_check(
+    shotover_peers: Vec<ShotoverNode>,
+    check_shotover_peers_delay_ms: u64,
+    connect_timeout_ms: u64,
+) {
+    if !shotover_peers.is_empty() {
+        tokio::spawn(async move {
+            // Wait for all shotover nodes to start
+            sleep(Duration::from_secs(10)).await;
+            loop {
+                match check_shotover_peers(
+                    &shotover_peers,
+                    check_shotover_peers_delay_ms,
+                    connect_timeout_ms,
+                )
+                .await
+                {
+                    Ok(_) => {}
+                    Err(err) => {
+                        tracing::error!(
+                            "Restarting the shotover peers check due to error: {err:?}"
+                        );
+                    }
+                };
+            }
+        });
+    }
+}
+
+async fn check_shotover_peers(
+    shotover_peers: &[ShotoverNode],
+    check_shotover_peers_delay_ms: u64,
+    connect_timeout_ms: u64,
+) -> Result<(), Error> {
+    let mut shotover_peers_cycle = shotover_peers.iter().cycle();
+    let mut rng = StdRng::from_rng(rand::thread_rng())?;
+    let check_shotover_peers_delay_ms = check_shotover_peers_delay_ms as i64;
+    loop {
+        if let Some(shotover_peer) = shotover_peers_cycle.next() {
+            let tcp_stream = tcp_stream(
```

Comment:
This can be written as:
```rust
            let tcp_stream = tcp_stream(
                Duration::from_millis(connect_timeout_ms),
                (
                    shotover_peer.address.host.as_str(),
                    shotover_peer.address.port as u16,
                ),
            )
            .await;
```
Which will avoid the cost of allocation of the string here.

Reply:
Nice